### PR TITLE
encode_tls_data: be more nice to allocator

### PR DESCRIPTION
**Pull request by hannesm**

### CODE REVIEW

**Review by reynir — APPROVED** [review]

File: src/packet.ml [hunk 1]
```diff
@@ -315,17 +315,26 @@ let encode_tls_data t =
...
   in
-  Cstruct.concat [ prefix; key_source; opt; u_p ]
+  (* prefix - 4 zero bytes, key_method
+     pre_master
+     random1
+     random2
+     opt string
+     user string
+     password string
+     NCP
```
Comment (reynir) [inline src/packet.ml:334]:
```suggestion
     peer-info
```

File: src/packet.ml [hunk 2]
```diff
@@ -315,17 +315,26 @@ let encode_tls_data t =
...
+     let key_source = Cstruct.concat [  ] in
```
Comment (reynir) [inline src/packet.ml:335]: I don't understand this comment

File: src/packet.ml (hunk 1 above)
Comment (hannesm) [reply src/packet.ml]: indeed. done in c0f72a8

File: src/packet.ml (hunk 2 above)
Comment (hannesm) [reply src/packet.ml]: leftover, removed in c0f72a8

**Pull request merged**